File: ruby/raindrops/raindrops.rb
class Raindrops
  KEYS = {
    3 => "Pling",
    5 => "Plang",
    7 => "Plong"
  }.freeze

  def self.convert(num)
    factors = (1..num).
      select { |n|num % n == 0 && [3,5,7].include?(n) }

    if factors.empty?
      num.to_s
    else
      factors.map { |f| Raindrops::KEYS[f] }.join("")
    end
  end
end

File: go/raindrops/raindrops.go
package raindrops

import "fmt"

const testVersion = 2

func Convert(num int) string {
	output := ""

	if num%3 == 0 {
		output += "Pling"
	}

	if num%5 == 0 {
		output += "Plang"
	}

	if num%7 == 0 {
		output += "Plong"
	}

	if output != "" {
		return output
	} else {
		return fmt.Sprintf("%d", num)
	}
}
